gm4_spawner_minecarts/data/gm4_spawner_minecarts/function/main.mcfunction

```diff
@@ -1,21 +1,7 @@
-# check for minecarts under spawners
-execute as @e[type=minecart,tag=!smithed.entity] at @s if block ~ ~1 ~ spawner align xyz positioned ~0.5 ~1.5 ~0.5 unless entity @e[tag=smithed.block,distance=..0.5] at @s if block ~ ~2 ~ #gm4_spawner_minecarts:piston[facing=down] if block ~ ~3 ~ redstone_block run function gm4_spawner_minecarts:capture_spawner/enter_minecart
-
-# slow clock for passive regeneration
-scoreboard players add passive_clock gm4_spawner_fuel 1
-execute if score passive_clock gm4_spawner_fuel matches 750.. run scoreboard players set passive_clock gm4_spawner_fuel 0
-
-# slow clock for active regeneration
-scoreboard players add active_clock gm4_spawner_fuel 1
-execute if score active_clock gm4_spawner_fuel matches 2.. run scoreboard players set active_clock gm4_spawner_fuel 0
+schedule function gm4_spawner_minecarts:main 16t
 
-# global timer
-execute if score passive_clock gm4_spawner_fuel matches 0 run scoreboard players add spawner_global_time gm4_spawner_time 1
+# check for minecarts under spawners
+execute as @e[type=minecart,tag=!smithed.entity] at @s if block ~ ~1 ~ spawner align xyz positioned ~0.5 ~1.5 ~0.5 unless entity @e[tag=smithed.block,distance=..0.5] at @s if block ~ ~2 ~ #gm4_spawner_minecarts:piston[facing=down] if block ~ ~3 ~ redstone_block run function gm4_spawner_minecarts:capture_spawner
```

Comment:
Could use a `limit=1` on the smithed block `unless entity` check. Just in case there are a bunch of entities within 0.5 blocks. In practice probably doesn't matter.